src/text_shaper/directwrite_shaper.cpp

```diff
@@ -671,42 +727,76 @@ std::optional<rasterized_glyph> directwrite_shaper::rasterize(glyph_key _glyph,
 
     auto const width = output.size.width;
     auto const height = output.size.height;
-    output.bitmap.resize(*height * *width * 3);
-    output.format = bitmap_format::rgb;
 
-    std::vector<uint8_t> tmp;
-    tmp.resize(*height * *width * 3);
+    IDWriteFactory2* factory2 = nullptr;
+    IDWriteColorGlyphRunEnumerator* glyphRunEnumerator;
+    if (SUCCEEDED(d->factory->QueryInterface(IID_PPV_ARGS(&factory2))))
+    {
+        hr = factory2->TranslateColorGlyphRun(
+            0.0f,
+            0.0f,
+            &glyphRun,
+            nullptr,
+            DWRITE_MEASURING_MODE::DWRITE_MEASURING_MODE_NATURAL,
+            nullptr,
+            0,
+            &glyphRunEnumerator);
 
-    hr = glyphAnalysis->CreateAlphaTexture(
-        DWRITE_TEXTURE_CLEARTYPE_3x1,
-        &textureBounds,
-        tmp.data(),
-        tmp.size()
-    );
+        if (hr == DWRITE_E_NOCOLOR)
+        {
+            output.bitmap.resize(*height * *width * 3);
+            output.format = bitmap_format::rgb;
 
-    auto t = output.bitmap.begin();
+            auto t = output.bitmap.begin();
 
-    for (auto i = 0; i < *height; i++)
-        for (auto j = 0; j < *width; j++)
-        {
-            const auto base = ((*height - 1 - i) * *width + j) * 3;
+            renderGlyphRunToBitmap(glyphAnalysis.Get(), textureBounds, output.format, t);
 
-            *t++ = tmp[base];
-            *t++ = tmp[base + 1];
-            *t++ = tmp[base + 2];
+            return output;
         }
+        else
+        {
+            output.bitmap.resize(*height * *width * 4);
+            output.format = bitmap_format::rgba;
 
-    if (FAILED(hr)) {
-        return nullopt;
+            d->fontsHasColor.at(_glyph.font) = true;
+            while (true)
+            {
+                BOOL haveRun;
+                glyphRunEnumerator->MoveNext(&haveRun);
```

Comment:
See https://docs.microsoft.com/en-us/windows/win32/directwrite/color-fonts#using-color-fonts-with-directwrite-and-direct2d.

This *should* give us colored bitmap, unless i am doing something wrong.